x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformTask.java

```diff
@@ -299,7 +295,7 @@ void setShouldStopAtCheckpoint(boolean shouldStopAtCheckpoint) {
      * @param shouldStopAtCheckpoint whether or not we should stop at the next checkpoint or not
      * @param shouldStopAtCheckpointListener the listener to return to when we have persisted the updated value to the state index.
      */
-    public synchronized void setShouldStopAtCheckpoint(
+    public void setShouldStopAtCheckpoint(
```

Comment:
Since this is no longer synchronized, it is possible for a `stop` call or a `start` call to be executing at any point. 

Is this acceptable?

Reply:
with the atomic ops in `stopAtCheckpoint`, I think we can get rid of _synchronized_. I can not think of any bad collision with `stop` and `start`.

Reply:
FWIW: I added _synchronized_ back, but 1 level down in the indexer, as I realized a corner case: >1 calls to at the same time or triggering while this is running.